tests/scripts/task_build.sh

```diff
@@ -15,5 +15,41 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-export VTA_HW_PATH=`pwd`/3rdparty/vta-hw
-cd $1 && cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo && make $2 && cd ..
+set -eux
+
+VTA_HW_PATH=$(pwd)/3rdparty/vta-hw
+export VTA_HW_PATH
+
+pushd "$1"
+
+# Set up sccache to use S3
+if [ -n "${3+x}" ] && which sccache; then
```

Comment:
should we error here if `which sccache` is empty?

Reply:
No since this is supposed to work even if sccache is not present

Reply:
i see, what i'm worried about is a performance regression if someone accidentally removes sccache from CI and then not catching it. it seems like this should maybe be like

```
if [ -n"${3+x}" ]; then
  sccache_location=$(which sccache)
  if [ -n "${CI+x}" -a "x${sccache_location}" == "x" ]; then
    echo "expected sccache but it was not found; aborting since build is running in CI"
    exit 2
  fi
fi
```

and then we should manually mark which images we expect it as a hack til we rebuild them all. wdyt?

Reply:
makes sense, I'd be more inclined to leave that to a follow up though since it'll involve some other changes because the CI scripts themselves aren't aware of what image they're run in so we'd need to thread that through. it should be simple though and we'll be closely monitoring build jobs in the meantime anyways